mlir/test/lib/Dialect/SCF/TestLoopUnrolling.cpp

```diff
@@ -77,6 +82,9 @@ struct TestLoopUnrollingPass
                                 llvm::cl::init(false)};
   Option<unsigned> loopDepth{*this, "loop-depth", llvm::cl::desc("Loop depth."),
                              llvm::cl::init(0)};
+  Option<bool> unrollFull{*this, "unroll-full",
+                          llvm::cl::desc("Full unroll loops."),
+                          llvm::cl::init(false)};
```

Comment:
Nit: I'd just make the unroll-factor take -1 as "full" and avoid potential conflict when passing both an explicit factor and unroll-full=true.

Reply:
Although this may sound like common advice, I think you are a genius.

Reply:
It's okay to change it from `uint64_t` to `int64_t`, but I won't insist. This is a test pass.

Reply:
For now, since `affine-loop-unroll` and `affine-loop-unroll-jam` also have this option, it's not too late to change it when it's not a test pass.